docker/base.dockerfile

```diff
@@ -14,10 +14,13 @@ ENV FAASM_DOCKER="on"
 # Copy Python runtime libraries
 COPY --from=python /usr/local/faasm/runtime_root /usr/local/faasm/runtime_root
 
-# Set up SGX SDK
+# Set up SGX SDK, and install the SDK and PSW libraries
 COPY --from=sgx /opt/intel /opt/intel
-RUN apt-get update
-RUN apt-get install -y nasm
+COPY --from=sgx /linux-sgx/linux/installer/bin/ /tmp/linux-sgx
+WORKDIR /opt/intel
+RUN sh -c 'echo yes | /tmp/linux-sgx/sgx_linux_x64_sdk_2.15.101.1.bin'
+RUN sh -c 'echo yes | /tmp/linux-sgx/sgx_linux_x64_psw_2.15.101.1.bin --no-start-aesm'
+RUN cp /opt/intel/sgxsdk/lib64/libsgx_capable.so /usr/lib
```

Comment:
This looks like SGX setup that should be included in the `sgx` base image, why does it need to be here?

Also why do we need to copy `/opt/intel/sgxsdk/lib64/libsgx_capable.so` to `/usr/lib`? I would expect `FindSGX.cmake` to still exist, find this library where it's installed, then set this as part of an `SGX_LIBS` variable that things could link with.

Reply:
This is a bit involved, and I tried to clarify it in comments. I will answer questions one by one:

> This looks like SGX setup that should be included in the sgx base image, why does it need to be here?

The second installer (PSW) installs some libraries under `/usr/lib`. Currently, in `base.dockerfile` we inherit from `cpp-sysroot` and just _copy_ the `/opt/intel` directory from the sgx image, missing the libraries installed in `/usr/lib`, needed in [HW mode](https://github.com/intel/linux-sgx/issues/47).

> Also why do we need to copy `/opt/intel/sgxsdk/lib64/libsgx_capable.so` to `/usr/lib`?

We need to link against this library both in `"Hardware"` and `"Simulation"` builds, as it helps detecting the status of SGX. Unfortunately, it is not installed to `/usr/lib`, only to `/opt/intel/sgxsdk/lib64/libsgx_capable.so`. Note that the libraries installed under `/usr/lib` also live in `/opt/intel/sgxsdk/lib64`. As a consequence, if `target_link_libraries`-ing to `libsgx_capable.so` in the latter path, the duplicated libraries in the former path (`/usr/lib`) will be shadowed by the ones in the latter (`/opt/intel/sgxsdk/lib64`), yielding the same error as before.

Reply:
Ok great, mentioning this in a comment in the code to explain would be good so that someone doesn't attempt to move it into the `sgx` container (which would have been my gut reaction).